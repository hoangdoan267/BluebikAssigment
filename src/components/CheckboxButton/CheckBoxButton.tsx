import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import React from 'react';
import {Colors} from '../../utils/color';

interface ICheckboxButton {
  label: string;
  value: string;
  isChosen: boolean;
  onPress: (value: string) => void;
}

const CheckBoxButton = (props: ICheckboxButton) => {
  const {label, isChosen, onPress, value} = props;

  const onPressCheckbox = () => {
    onPress(value);
  };

  return (
    <TouchableOpacity
      testID="chooseButton"
      style={styles.container}
      onPress={onPressCheckbox}
      activeOpacity={0.9}>
      <View
        style={[
          styles.outerBox,
          {
            borderColor: isChosen
              ? Colors.BGSupport.Primary
              : Colors.TextBlack.Teritary,
          },
        ]}>
        {isChosen && <View style={styles.innerBox} />}
      </View>
      <Text
        style={[
          styles.label,
          {
            color: isChosen
              ? Colors.BGSupport.Primary
              : Colors.TextBlack.Teritary,
          },
        ]}>
        {label}
      </Text>
    </TouchableOpacity>
  );
};

export default CheckBoxButton;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: '100%',
    alignItems: 'center',
    marginBottom: 8,
  },
  label: {
    fontSize: 18,
    color: Colors.TextBlack.Teritary,
    fontWeight: 'bold',
  },
  outerBox: {
    width: 18,
    height: 18,
    borderRadius: 4,
    borderWidth: 1,
    marginRight: 4,
    alignItems: 'center',
    justifyContent: 'center',
  },
  innerBox: {
    width: 10,
    height: 10,
    borderRadius: 2,
    backgroundColor: Colors.BGSupport.Primary,
  },
});
