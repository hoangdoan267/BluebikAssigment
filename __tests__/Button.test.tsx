import * as React from 'react';
import CheckBoxButton from '../src/components/CheckboxButton/CheckBoxButton';
import {act, fireEvent, render} from '@testing-library/react-native';

describe('CheckBoxButton suites', () => {
  const onPress = jest.fn();
  test('it render isChosen = true correctly', () => {
    const button = render(
      <CheckBoxButton
        label="default checkbox"
        value="default"
        isChosen
        onPress={onPress}
      />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
  test('it render isChosen = false correctly', () => {
    const button = render(
      <CheckBoxButton
        label="default checkbox"
        value="default"
        isChosen={false}
        onPress={onPress}
      />,
    ).toJSON();
    expect(button).toMatchSnapshot();
  });
  test('it onPress called', async () => {
    const button = render(
      <CheckBoxButton
        label="default checkbox"
        value="default"
        isChosen={false}
        onPress={onPress}
      />,
    );
    const onClickButton = button.getByTestId('chooseButton');
    await act(() => {
      fireEvent.press(onClickButton);
    });
    expect(onPress).toBeCalled();
  });
});
