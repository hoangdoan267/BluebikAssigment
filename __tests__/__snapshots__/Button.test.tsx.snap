// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxButton suites it render isChosen = false correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 8,
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="chooseButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 18,
          "justifyContent": "center",
          "marginRight": 4,
          "width": 18,
        },
        {
          "borderColor": "rgba(12, 12, 14, 0.45)",
        },
      ]
    }
  />
  <Text
    style={
      [
        {
          "color": "rgba(12, 12, 14, 0.45)",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        {
          "color": "rgba(12, 12, 14, 0.45)",
        },
      ]
    }
  >
    default checkbox
  </Text>
</View>
`;

exports[`CheckBoxButton suites it render isChosen = true correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 8,
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="chooseButton"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "height": 18,
          "justifyContent": "center",
          "marginRight": 4,
          "width": 18,
        },
        {
          "borderColor": "#6A00C8",
        },
      ]
    }
  >
    <View
      style={
        {
          "backgroundColor": "#6A00C8",
          "borderRadius": 2,
          "height": 10,
          "width": 10,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "rgba(12, 12, 14, 0.45)",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        {
          "color": "#6A00C8",
        },
      ]
    }
  >
    default checkbox
  </Text>
</View>
`;
